src/modules/job-manager/plugins/dependency-after.c

```diff
@@ -0,0 +1,563 @@
+     */
+    if (after_type_parse (scheme, &type) < 0)
+        return flux_jobtap_reject_job (p,
+                                       args,
+                                       "invalid dependency scheme: %s",
+                                       scheme);
+
+    /*  Parse the value argument, which must be a valid jobid
+     *  Do not allow FLUX_JOBID_ANY/FLUX_JOBTAP_CURRENT_JOBID to be specified
+     */
+    if (flux_job_id_parse (jobid, &afterid) < 0
+        || afterid == FLUX_JOBTAP_CURRENT_JOB)
+        return flux_jobtap_reject_job (p, args,
+                                       "%s: %s: \"%s\" is not a valid jobid",
+                                       "dependency",
+                                       scheme,
+                                       jobid);
+
+    /*  Lookup userid and state of target job `afterid`
+     */
+    if (lookup_job_uid_state (p, afterid, &target_uid, &target_state) < 0) {
+        return flux_jobtap_reject_job (p, args,
+                                       "%s: %s: id %s: %s",
+                                       "dependency",
+                                       scheme,
+                                       jobid,
+                                       errno == ENOENT ?
+                                       "job not found" :
+                                       strerror (errno));
+    }
+
+    /*  Requesting userid must match target job uid
+     */
+    if (uid != target_uid)
+        return flux_jobtap_reject_job (p,
+                                       args,
+                                       "%s: Permission denied for job %s",
+                                       scheme,
+                                       jobid);
+
+    if (!(after = after_info_create (id, type, jobid)))
+        return flux_jobtap_reject_job (p, args,
+                                       "failed to establish job dependency");
+
+    /*  Emit the dependency
+     */
+    if (flux_jobtap_dependency_add (p, id, after->description) < 0) {
+        after_info_destroy (after);
+        return flux_jobtap_reject_job (p, args, "Unable to add job dependency");
+    }
+
+    /*  Corner case, requisite job may have already started. Check for that
+     *   here and immediately satisfy dependency before adding to various
+     *   lists below
+     */
+    if (type == AFTER_START
+        && (target_state == FLUX_JOB_STATE_RUN
+            || target_state == FLUX_JOB_STATE_CLEANUP
+            || target_state == FLUX_JOB_STATE_INACTIVE)) {
+        if (flux_jobtap_dependency_remove (p, id, after->description) < 0)
+            flux_log_error (flux_jobtap_get_flux (p),
+                            "flux_jobtap_dependency_remove");
+        after_info_destroy (after);
+        return 0;
+    }
+
+    /*  Append this dependency to the deplist in the target jobid:
+     */
+    if (!(l = after_list_get (p, afterid))
+        || !(handle = zlistx_add_end (l, after))) {
```

Comment:
`handle` assigned but not used

Reply:
Ah, good catch. In previous versions handle was used in the reference but then I remembered use of zlistx handles is not safe since they are re-assigned on a sort.